server/channels/api4/report.go

```diff
@@ -85,3 +88,46 @@ func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Logger.Warn("Error writing response", mlog.Err(jsonErr))
 	}
 }
+
+func startUsersBatchExport(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
```

Comment:
We already do this with all of the other reporting calls so I wanted to keep consistent. I know there's some way you can give users access to the admin console without being a system admin from what I can tell, so I wanted to cover both for this function and decide later if we want to loosen the restriction. On top of that, adding an extra check for security doesn't hurt too much.

Reply:
> so I wanted to cover both for this function and decide later if we want to loosen the restriction.

At a minimum, let's write the permission check to reflect the current requirements -- which is being a system admin -- and add code later to check for a specific permission if we want to loosen the restriction. Or better yet, introduce a new permission to cover this case and only grant it to system administrators for now (leaving it up to advanced permissions to customize).

What we have right now really muddies the permission check, making it hard to read and immediately grasp what's going on, since the `model.PermissionSysconsoleReadUserManagementUsers` is illusory.